types/react-dom/client.d.ts

```diff
@@ -35,6 +35,6 @@ export function createRoot(container: Element | DocumentFragment, options?: Root
 
 export function hydrateRoot(
     container: Element | Document,
-    initialChildren: React.ReactChild | Iterable<React.ReactNode>,
+    initialChildren: React.ReactNode,
```

Comment:
https://github.com/facebook/react/blob/fe6e0741286345edb2aa23784c21f8ea611ebdea/packages/react-dom/src/client/ReactDOMRoot.js#L257
`ReactNodeList` is our `ReactNode`